domain-ee/ee-max-cfi-app/scripts/dc_lookup_table_updater.py

```diff
@@ -0,0 +1,75 @@
+import csv
+import sys
+from typing import Any
+
+DATA_FILE = '../src/python_src/util/data/Diagnostic Code Lookup Table.csv'
+
+
+def consolidate_data_files(new_data_file: str) -> dict[int, tuple[Any, Any, Any, Any, Any, Any]] | None:
+    original_data = import_file(DATA_FILE)
+    data_map: dict[int, tuple[Any, int, Any, Any, Any, Any]] = original_data.copy()
+
+    imported_data = import_file(new_data_file)
+    updates = imported_data & data_map.keys()
+    adds = updates ^ imported_data.keys()
+    msg = f'Loading {len(imported_data)} ratings from "{new_data_file}" | Adds {len(adds)} | Potential Updates {len(updates)}'
+    print(msg)
+
+    data_map.update(imported_data)
+
+    new_data = dict(sorted(data_map.items()))
+
+    if new_data == original_data:
+        print('No changes detected')
+        return None
+
+    return new_data
+
+
+def import_file(filename: str) -> dict[int, tuple[Any, Any, Any, Any, Any, Any]]:
+    diagnostic_code_to_data: dict[int, tuple[Any, Any, Any, Any, Any, Any]] = {}
+    with open(filename, 'r') as file:
+        csv_reader = csv.reader(file)
+        for index, csv_line in enumerate(csv_reader):
+            if index == 0:
+                continue
+
+            if len(csv_line) != 7:
+                raise ValueError(f'Invalid CSV line at index {index}')
+
+            diagnostic_code_str, rated_issue_name, max_rating_str, body_system, category, subcategory, cfr_ref = csv_line
+
+            try:
+                diagnostic_code = int(diagnostic_code_str)
+            except ValueError:
+                raise ValueError(f'Invalid diagnostic code at index {index}: \n{csv_line}')
+
+            diagnostic_code_to_data[diagnostic_code] = (rated_issue_name, max_rating_str, body_system, category, subcategory, cfr_ref)
+
+    return diagnostic_code_to_data
+
+
+def export_data(data: dict[int, tuple[Any, int, Any, Any, Any, Any]]):
+    with open(DATA_FILE, 'w') as file:
+        csv_writer = csv.writer(file)
+        csv_writer.writerow(['Diagnostic Code', 'Rated Issue Name', 'Max Rating', 'Body System', 'Category', 'Subcategory', 'CFR Reference'])
+        for diagnostic_code, data_tuple in data.items():
+            csv_writer.writerow([diagnostic_code, *data_tuple])
+
+
+def increment_table_version():
+    with open('../src/python_src/util/data/table_version.py', 'r') as file:
+        version = int(file.readline().strip().split(' = ').pop())
```

Comment:
The brittleness of this line makes me a bit nervous, because at a glance `table_version.py` is just any Python file, but it has to be formatted a certain way or else this function breaks.

Suggestions:
- You could convert it into a minimal text file with just the version number, which makes this parsing simpler, but instead of importing it later you'd have to read-and-parse.
- Or you could bank on the `readline()` behavior and add a human-readable comment at line 2 saying it's auto-generated by `dc_lookup_table_updater.py`.